### PR TITLE
Estonian 6.11.1: XTE-99, XTE-252, XTE-293, XTE-294, XTE-307, XTE-308, XTE-310, XTE-333

### PR DESCRIPTION
**Pull request by kristoheero**
* XTE-99 / Joint development issue #79: Security Server UI: Added uniqueness check of the entered security server code when initializing the server.
 * XTE-252 / Joint development issue #53: Security Server: Upgraded embedded Jetty to the version 9.4.2.
 * XTE-293: Security Server: A field set used to generate the token ID of the SSCD has been made configurable.
* XTE-294 / Joint development issue #84: Security Server: Added configuration file for the OCSP responder Jetty server (and increased max threads size of the thread pool).
 * XTE-307 / Joint development issue #131: Security Server bugfix: Added missing HTTP header "Connection: close" into the server proxy response in cases error occurs before parsing a service provider's response.
 * XTE-308 / Joint development issue #132: Security Server bugfix: Added missing read timeout for OCSP responder client.
 * XTE-310 / Joint development issue #125: Security Server bugfix: SOAP messages with attachments caused in some cases a temopray file handle leak.
 * XTE-333 / Joint development issue #128: Security Server bugfix: Fixed parsing SOAP messages containing &amp; or &lt; entities.
 * Security Server: TCP socket SO_LINGER values in the proxy configuration file (proxy.ini) set to -1 according to avoid unexpected data stream closures.

Also updated documents:

* IG-SS
* UG-SYSPAR

### CODE REVIEW

File: src/proxy/src/main/java/ee/ria/xroad/proxy/protocol/ProxyMessage.java [hunk 1]
```diff
@@ -111,8 +112,9 @@ public SoapFault getFault() {
      */
     public String getSoapContentType() {
         return isMimeEncodedSoap() || hasAttachments()
-                ? (originalContentType != null ? originalContentType : encoder.getContentType())
-                : MimeUtils.TEXT_XML_UTF8;
+                ? originalContentType != null
+                    ? originalContentType : encoder.getContentType()
+                : MimeTypes.TEXT_XML_UTF8;
     }
```
Comment (olli-lindgren) [inline src/proxy/src/main/java/ee/ria/xroad/proxy/protocol/ProxyMessage.java:118]: Was this broken? Is the new version easier to read? This seems like it should have at least one less ternary operator.

**Review by olli-lindgren — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
It is a big pull request and it is hard to say which modifications exactly perform the fix for https://github.com/vrk-kpa/xroad-joint-development/issues/128 that we are looking for. The commit history does not help (have you rebased it or why do the individual commits not show?) Can you make a separate pull request for that individual fix?

**Comment by iluwatar** [comment]
The version number indicates that the pull request contains only fixes on top of 6.11.0? Is this really the case or should the version number be 6.12.0 because features have been added?

**Comment by jhyoty** [comment]
Fix for XTE-333 works only partially; entities in attributes (`<element attr="&amp;value">`) are not handled correctly.

**Review by iluwatar — CHANGES_REQUESTED** [review]
We request changes because the fix for https://github.com/vrk-kpa/xroad-joint-development/issues/128 only seems to work partially and it is hard to say which modifications exactly belong to that bug fix.

File: src/proxy/src/main/java/ee/ria/xroad/proxy/protocol/ProxyMessage.java (hunk 1 above)
Comment (kristoheero) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/protocol/ProxyMessage.java:118]: Well, historically we are using still code lines with length 80 character, but X-Road check style uses currently 120 length and indeed, no wrapping here is needed.  Old habits :). We will revert it.

**Comment by kristoheero** [comment]
>It is a big pull request and it is hard to say which modifications exactly perform the fix for vrk-kpa/xroad-joint-development#128 that we are looking for. The commit history does not help (have you rebased it or why do the individual commits not show?) Can you make a separate pull request for that individual fix?

With #128 are involved files:
* src/common-util/build.gradle
* src/common-util/src/main/java/ee/ria/xroad/common/message/SaxSoapParserImpl.java
* src/common-util/src/main/java/ee/ria/xroad/common/util/AbstractHttpSender.java 
* src/proxy/src/test/java/ee/ria/xroad/proxy/testsuite/testcases/XmlEncodedMessage.java 
* src/proxy/src/test/queries/xmlencoded.answer
* src/proxy/src/test/queries/xmlencoded.query

Currently we are merge-squashing and anonymising internal commits before pull request.

**Comment by VitaliStupin** [comment]
I think change log should contain information that due to upgrade of embedded Jetty SHA1 ciphers are no longer supported for communication between security server and client. And I agree that upgrade of jetty is more feature then bug fix. Also https://github.com/vrk-kpa/xroad-joint-development/issues/79 is a feature. So the new version could really be 6.12.0.

**Comment by kristoheero** [comment]
I made new pull request for _estonian-6.12.0_ which contains additional commits.

**Pull request closed without merging**